File: manual/code/test_suites/python/python/test_sortedcontainers_SortedList_remove.py
import unittest
from function_sortedcontainers_SortedList_remove import HanSeok

class TestSortedListRemoveMethod(unittest.TestCase):

    def test_remove_element(self):
        sorted_list = HanSeok().remove_element_from_list(3)
        self.assertEqual(sorted_list, "1,5,8")

    def test_remove_element_not_found(self):
        result = HanSeok().remove_element_from_list(10)
        self.assertEqual(result, "Value not found")

    def test_remove_duplicate_element(self):
        sorted_list = HanSeok().remove_element_from_list(3, "1,3,3,5")
        self.assertEqual(sorted_list, "1,3,5")

    def test_remove_all_elements(self):
        sorted_list = "1,3,5,8"
        for value in sorted_list.split(","):
            sorted_list = HanSeok().remove_element_from_list(int(value), sorted_list)
        self.assertEqual(sorted_list, "")

    def test_empty_list(self):
        result = HanSeok().remove_element_from_list(1, "1")
        self.assertEqual(result, "")

if __name__ == "__main__":
    unittest.main()

File: manual/data/cpp/tests/test_sortedcontainers_SortedList_remove.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/function_sortedcontainers_SortedList_remove.cpp"

class TestSortedListRemoveMethod : public ::testing::Test {
protected:
    HanSeok handler;
};

TEST_F(TestSortedListRemoveMethod, test_remove_element) {
    std::string result = handler.remove_element_from_list(3);
    EXPECT_EQ(result, "1,5,8");
}

TEST_F(TestSortedListRemoveMethod, test_remove_element_not_found) {
    std::string result = handler.remove_element_from_list(10);
    EXPECT_EQ(result, "Value not found");
}

TEST_F(TestSortedListRemoveMethod, test_remove_duplicate_element) {
    std::string result = handler.remove_element_from_list(3, "1,3,3,5");
    EXPECT_EQ(result, "1,3,5");
}

TEST_F(TestSortedListRemoveMethod, test_remove_all_elements) {
    std::string sorted_list = "1,3,5,8";
    std::vector<int> values;
    std::istringstream ss(sorted_list);
    std::string token;
    
    while (std::getline(ss, token, ',')) {
        if (!token.empty()) {
            values.push_back(std::stoi(token));
        }
    }
    
    for (const auto& value : values) {
        sorted_list = handler.remove_element_from_list(value, sorted_list);
    }
    
    EXPECT_EQ(sorted_list, "");
}

TEST_F(TestSortedListRemoveMethod, test_empty_list) {
    std::string result = handler.remove_element_from_list(1, "1");
    EXPECT_EQ(result, "");
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}